### ISSUE TITLE
Disabled dynamic fields shown in CustomerTicketZoom

### ISSUE THREAD

**Issue by cugu** [RotherOSS/otobo#1447]
Hi,

The CustomerTicketZoom view shows the fields "process" and "activity" even when they are disabled in the screens.

**Comment by reneeb** [comment]
Process and Activity are special as they are always shown in process tickets:

see lines 1713 .. 
https://github.com/RotherOSS/otobo/blob/163c96b00a62d38e5509d3e595c69e256537a51d/Kernel/Modules/CustomerTicketZoom.pm#L1713

@svenoe This could be configurable as those information can "leak" internal organisational data to customers...

**Comment by svenoe** [comment]
I agree. I have not looked into where exactly the info is shown, but if CustomerTicketZoom###DynamicField can be used here, in a sensible way, we should do that, instead of introducing new options. I don't know whether we have two separate places, though. (Maybe we can just disable one, then.)

**Comment by reneeb** [comment]
I'll take a closer look at it at the end of the week.

**Comment by stefanhaerter** [comment]
So far it appears to be the case that, if Process and Activity are enabled vor CustomerTicketZoom, each appears twice in the information widget. This is due to the fact that they are rendered once via the dynamic fields handling and once via the code marked above. So I would suggest to remove the code fragment in Kernel/Modules/CustomerTicketZoom.p, L.1713ff, but before it has to be checked if the handling in Kernel/System/DynamicField/Driver/ProvessManagement/ActivityID.pm and ProcessID.pm is correct.

Another problem has occured to me while investigating this issue. When reaching https://github.com/RotherOSS/otobo/blob/rel-10_1/Kernel/System/ProcessManagement/Activity.pm#L100, some activity entities appear to be broken, like the second item here:

```
$Activity = {
              "Activity-991d30fdf276f4344bc5b608c24c1b60" => {
                                                               ActivityDialog => { 1 => "ActivityDialog-a66dedc3b5f7213496d52550c0ebef93" },
                                                               ChangeTime => "2022-07-06 06:59:28",
                                                               CreateTime => "2022-07-06 06:59:28",
                                                               ID => 1,
                                                               Name => "TestAktivit\xE4t1",
                                                             },     
              "Activity-e2712e2c4433f53e745b48d94b09a2c3" => {
                                                               1 => "ActivityDialog-967df513b3b04e8e89b0777c0bcc3313",
                                                               ActivityDialog => {}, 
                                                             },     
            }    
```  

Till now, I couldn't figure out how this happens. In the Database as well as in Kernel/Config/Files/ZZZProcessmanagement.pm, the data appears to be correct. I am currently investigating https://github.com/RotherOSS/otobo/blob/rel-10_1/Kernel/Config/Defaults.pm#L2246.

**Comment by stefanhaerter** [comment]
Had a thought and wanted to leave this here: How about leaving the hardcoded process and activity information as it is and declaring the corresponding dynamic fields as internal fields, like e.g. ITSMCriticality and ITSMImpact?

**Comment by stefanhaerter** [comment]
Restarting to take on this issue. As this is a change of behavior, I'll implement it based on `rel-11_1` and if needed, it can be backported.

**Comment by stefanhaerter** [comment]
Following a brief discussion, the decision is to enable the two dynamic fields for CustomerTicketZoom and AgentTicketZoom per default as well as providing a script to enable them during upgrades to ensure continuity.

**Comment by stefanhaerter** [comment]
PR merged, closing the issue.

**Comment by bschmalhofer** [comment]
While testing the migration from OTOBO 11.0.x to 11.1.x I noticed a problem in the migration task SysConfigUpdateTicketZoomDFScreens . Two config settings had no value as these were new settings and Maint::Config::Rebuild had not yet been executed. See

[otobo/scripts/DBUpdateTo11_1/SysConfigUpdateTicketZoomDFScreens.pm](https://github.com/RotherOSS/otobo/blob/de31d2fcad2f0eb09d4cb539eba86613a3c829ea/scripts/DBUpdateTo11_1/SysConfigUpdateTicketZoomDFScreens.pm#L52)

Line 52 in [de31d2f](https://github.com/RotherOSS/otobo/commit/de31d2fcad2f0eb09d4cb539eba86613a3c829ea)
 my $ProcessIDDF  = $ConfigObject->Get('Process::DynamicFieldProcessManagementProcessID'); 

TODO:

    Run Maint::Config::Rebuild at the start of the task

Another option is to have Maint::Config::Rebuild as a separate migration task that runs before the other tasks. But that is a different issue.
Markdown Editor
Markdown input: edit mode selected.
